### PR TITLE
Add more features to chat

### PR DESCRIPTION
**Pull request by jmbejar**
Among other things, we are adding the ability to delete and edit messages.

### CODE REVIEW

File: moxin-frontend/src/chat/chat_panel.rs [hunk 1]
```diff
@@ -254,8 +248,8 @@ pub struct ChatPanel {
     #[rust]
     auto_scroll_cancellable: bool,
 
-    #[rust(true)]
-    prompt_enabled: bool,
+    #[rust]
+    is_chat_streaming: bool,
```
Comment (fmzbl) [inline moxin-frontend/src/chat/chat_panel.rs:252]: (This may be out of the scope of the PR, we can discuss it later or smth, just a thought from a quick glance)

These 4 flags are starting to make the rest of the widget a bit messy and hard to follow, does it make sense to go with a sort of state machine approach here? maybe a full FSM is a bit of an overkill, but we could start encapsulating the state in an enum and have a couple of functions in the struct to handle stuff like the scrolling and state transitioning. Again, not wanting to make stuff overcomplicated here so lmk if im overthinking. 
```rust
enum ChatPanelState {
    Loading,
    Ready { auto_scroll_pending: bool, auto_scroll_cancellable: bool },
    Streaming,
}
```

**Comment by fmzbl** [comment]
hey! I wasn't able to run this because it doesn't find `ViewAction::FingerHoverIn` on the makepad-widgets package in the moxin-changes branch. But I'll comment on what i saw from a brief look.

File: moxin-frontend/src/chat/chat_panel.rs [hunk 2]
```diff
@@ -296,16 +290,30 @@ impl Widget for ChatPanel {
...
+                .chars()
+                .next()
+                .unwrap()
+                .to_uppercase()
+                .to_string();
         } else {
             chat_history = vec![];
             model_filename = "".to_string();
+            initial_letter = "".to_string();
         };
```
Comment (fmzbl) [inline moxin-frontend/src/chat/chat_panel.rs:308]: we could get rid of the mutable variables with the [map_or](https://doc.rust-lang.org/std/option/enum.Option.html#method.map_or) method, though im totally find with the imperative approach, its in most places in this code and i find it completely readable, so take this just as a curiosity.

```rust
let (chat_history, model_filename, initial_letter) =
    store
	.current_chat
	.as_ref()
	.map_or((vec![], "".to_string(), "".to_string()), |chat| {
	    let model_filename = chat.model_filename.clone();
	    let initial_letter = model_filename
		.chars()
		.next()
		.unwrap_or_default()
		.to_uppercase()
		.to_string();
	    (chat.messages.clone(), model_filename, initial_letter)
	});
```

**Review by fmzbl — COMMENTED** [review]
looks good! just a thought about the ChatPanel, which i wouldn't necessarily address it in this pr.

**Review by fmzbl — APPROVED** [review]

**Review by joulei — APPROVED** [review]

**Comment by jmbejar** [comment]
> hey! I wasn't able to run this because it doesn't find `ViewAction::FingerHoverIn` on the makepad-widgets package in the moxin-changes branch. I'll see if i can fix it, but I'll comment on what i saw from a brief look.

Cargo.lock changes should fix this
